src/storm/mesos/MesosNimbus.java

```diff
@@ -617,6 +608,56 @@ public void assignSlots(Topologies topologies, Map<String, Collection<WorkerSlot
       }
     }
   }
+  
+  private ExecutorInfo createExecutorInfo(Optional<ContainerInfo.Builder> containerInfoBuilder,
+          TopologyDetails details, String executorDataStr, double executorCpu, double executorMem, 
+          String cpuRole, String memRole, String configUri) {
+              
+        ExecutorInfo.Builder infoBuilder = ExecutorInfo.newBuilder()
+                .setExecutorId(ExecutorID.newBuilder().setValue(details.getId()))
+                .setData(ByteString.copyFromUtf8(executorDataStr))
+                .addResources(Resource.newBuilder()
```

Comment:
Ewww, awesome that you're already pulling this out into a separate function.  I'll make a separate PR to allow having the cpus and mem be 0.  i.e., you cannot add 0 as a resource since the mesos-master will reject it.  We are running with 0 CPUs for the executor because:
* if you dedicate a whole CPU then it's wasteful, the supervisor doesn't really do much
* if you dedicate a franctional CPU then the mesos UI and logs will get totally wonky since there's not great handling for rounding of fractional CPUs (using a resource of 0.1 leads to having available resources be like:  132.999125142456).  At least in mesos-0.20.1 which we are stuck on for now.
* CPUs are the precious resource in our system, we cannot afford to spend hundreds of CPUs on supervisors that do almost nothing.